SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularitySlaveAndRackManager.java

```diff
@@ -396,14 +397,19 @@ public void loadSlavesAndRacksFromMaster(MesosMasterStateObject state, boolean i
   }
 
   @Timed
-  public void checkOffer(Offer offer) {
+  public void checkOffer(Offer offer, Set<String> inactiveSlaves) {
     final String slaveId = offer.getSlaveId().getValue();
     final String rackId = slaveAndRackHelper.getRackIdOrDefault(offer);
     final String host = slaveAndRackHelper.getMaybeTruncatedHost(offer);
     final Map<String, String> textAttributes = slaveAndRackHelper.getTextAttributes(offer);
 
     final SingularitySlave slave = new SingularitySlave(slaveId, host, rackId, textAttributes, Optional.<MesosResourcesObject>absent());
 
+    if (inactiveSlaves.contains(host)) {
```

Comment:
maybe instead of doing this check here, we can do it after we find out if it's a new slave. My idea here would be that we can find the new slave, but if it is in the 'bad hosts' list, move it immediately to decommissioned. The end goal is to not schedule tasks on a slave that might be flapping due to issues. So, we definitely don't want it as active, but if we ignore it or mark it dead we don't have any info bout the fact that it's still (somewhat) present. Moving it to decommissioned right away (maybe with a message set) lets us know it's still there, but without the danger of scheduling tasks on a bad slave.